app/services/service_agreement_creator.rb

```diff
@@ -22,22 +24,22 @@ def call(service_agreement_params)
   private
 
   def modify_org_service_agreement_params(service_agreement_params)
-    civil_servant = civil_servant(service_agreement_params)
+    @civil_servant = find_or_create_civil_servant(service_agreement_params)
```

Comment:
Why do you reinitialize it here again with the same params from initialize? Also the method accepts an "email" as parameter but you pass "service_agreement_params"?